kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/InternalUtils.java

```diff
@@ -169,4 +168,11 @@ public static Path relativizePath(Path child, URI root) {
   public static Set<String> toLowerCaseSet(Collection<String> set) {
     return set.stream().map(String::toLowerCase).collect(Collectors.toSet());
   }
+
+  public static <T> List<T> toFilteredList(
+      CloseableIterator<T> iterator, Function<T, Boolean> filter) {
+    List<T> result = new ArrayList<>();
+    iterator.filter(filter).forEachRemaining(result::add);
+    return result;
+  }
```

Comment:
Sorry nit, I think it's easier if this is just `toList` and we can do the iterator filtering in the outer call. This way we can re-use this even when we don't want to filter

i.e. `toList(iterator.filter(filter))`